switmt_to_iso20022/utils.bal

```diff
@@ -0,0 +1,1440 @@
+# if not, it checks the second address array (`address2`). If neither is available,
+# it returns `null`. The function aggregates all address lines into a string array.
+#
+# + address1 - An optional array of `AdrsLine` that may contain address lines.
+# + address2 - An optional array of `AdrsLine` that may also contain address lines (default is `null`).
+# + return - Returns an array of strings representing the address lines if any address lines are found;
+# otherwise, returns `null`.
+isolated function getAddressLine(swiftmt:AdrsLine[]? address1, swiftmt:AdrsLine[]? address2 = ()) returns string[]? {
+    swiftmt:AdrsLine[] finalAddress;
+    if address1 is swiftmt:AdrsLine[] {
+        finalAddress = address1;
+    } else if address2 is swiftmt:AdrsLine[] {
+        finalAddress = address2;
+    } else {
+        return ();
+    }
+    return from swiftmt:AdrsLine adrsLine in finalAddress select adrsLine.content;
+}
+
+# Retrieves the details charges code based on the provided `Cd` code.
+# It looks up the code in an array and returns the corresponding details charge description.
+# If the code is not found, it returns an error.
+#
+# + code - An optional `Cd` object that contains the code to be looked up.
+# + return - Returns the details charge description associated with the provided code;
+# Otherwise an error.
+isolated function getDetailsChargesCd(swiftmt:Cd? code) returns string|error {
+    string[][] chargesCodeArray = DETAILS_CHRGS;
+    if code is swiftmt:Cd {
+        foreach string[] line in chargesCodeArray {
+            if line[0].equalsIgnoreCaseAscii(code.content) {
+                return line[1];
+            }
+        }
+        return error("Details of charges code is invalid.");
+    }
+    return error("Details of charges code is madatory.");
+}
+
+# Extracts and returns regulatory reporting details from the provided `MT77B` object.
+# The function parses the `Nrtv` content of the `MT77B` object to determine the regulatory reporting details
+# based on specific content patterns. It returns an array of `RgltryRptg` objects with the extracted details.
+#
+# + rgltyRptg - An optional `MT77B` object that contains the regulatory reporting information.
+# + return - Returns an array of `RgltryRptg` objects with the extracted regulatory reporting details.
+# The details are based on the content of the `Nrtv` field within the `MT77B` object.
+isolated function getRegulatoryReporting(string? rgltyRptg) returns isorecord:RegulatoryReporting3[]? {
+    if rgltyRptg is string {
+        if rgltyRptg.substring(1, 9).equalsIgnoreCaseAscii("BENEFRES") || 
+           rgltyRptg.substring(1, 9).equalsIgnoreCaseAscii("ORDERRES") {
+            string additionalInfo = "";
+            if rgltyRptg.length() > 14 {
+                foreach int i in 14 ... rgltyRptg.length()-1 {
+                    if rgltyRptg.substring(i,i+1).equalsIgnoreCaseAscii("/") {
```

Comment:
```suggestion
                    if rgltyRptg.substring(i, i + 1).equalsIgnoreCaseAscii("/") {
```

Reply:
There are formatting issues like this, please check other places too.

Reply:
fixed in `b55fa63`